lib/online_order.rb

```diff
@@ -0,0 +1,79 @@
+require 'csv'
+require_relative 'order'
+module Grocery
+  class OnlineOrder < Grocery::Order
+    attr_reader :customer, :status
+
+    def initialize (id, products, customer, status = :pending)
+      super(id, products)
+      @customer = customer
+      @status = status
+      valid_status = [:pending, :paid, :processing, :shipped, :complete]
+      if !valid_status.include?status
+        throw ArgumentError.new("Invalid status for online order '#{status}''")
+      end
+    end
+
+    def total
+      total = super
+      return total + 10
```

Comment:
Isn't `OnlineOrder` supposed to not add a shipping fee if there are no products?